### PR TITLE
Feature: APP-966 - Proposal Details - Voting Terminal

### PR DESCRIPTION
**Pull request by Fabricevladimir**
## Description
- Adds proposal settings and results to `VotingTerminal` on Proposal Details page
- Adds voting logic (voting eligibility is **mocked**)
- Sets `voting power` and `token amount` columns to optional in `VotersTable`

Task: [APP-966]()

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [x] I have selected the correct base branch.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [x] Any dependent changes have been merged and published in downstream modules.
- [ ] I ran all tests with success and extended them if necessary.
- [ ] I have updated the ``CHANGELOG.md`` file in the root folder.
- [x] I have tested my code on the test network.

### CODE REVIEW

**Review by RakeshUP — APPROVED** [review]

File: packages/web-app/src/pages/proposal.tsx [hunk 1]
```diff
@@ -82,6 +108,140 @@ const Proposal: React.FC = () => {
...
+          ? t('votingTerminal.status.voteSubmitted')
+          : t('votingTerminal.voteNow');
+        break;
+    }
+    return [voteStatus, voteButtonLabel];
+  }, [proposal?.endDate, proposal?.startDate, proposal?.status, t, voted]);
+
+  // vote button state and handler
+  const {voteNowDisabled, onClick} = useMemo(() => {
+    let voteNowDisabled = true;
```
Comment (WissenIstNacht) [inline packages/web-app/src/pages/proposal.tsx]: No need for function-wide temps. You could simply return from the various if/else blocks.

File: packages/web-app/src/pages/proposal.tsx [hunk 2]
```diff
@@ -95,7 +255,7 @@ const Proposal: React.FC = () => {
         {!isDesktop && (
           <Breadcrumb
             onClick={(path: string) =>
-              navigate(generatePath(path, {network, daoId}))
+              navigate(generatePath(path, {network, daoId: 'daoId'}))
```
Comment (WissenIstNacht) [inline packages/web-app/src/pages/proposal.tsx]: Why `daoId`?

File: packages/web-app/src/utils/proposals.ts [hunk 3]
```diff
@@ -0,0 +1,243 @@
...
+  totalWeight: bigint
+): {voters: Array<VoterType>; summary: string} {
+  let weights = BigNumber.from(0);
+  let votingPower;
+  let tokenAmount;
+
+  // map to voters structure
+  const voters = votes.map(vote => {
+    // increment number of voted tokens
+    weights = weights.add(BigNumber.from(vote.weight));
```
Comment (WissenIstNacht) [inline packages/web-app/src/utils/proposals.ts:87]: Is this not the same as total weights? Also, it's not used...?

File: packages/web-app/src/utils/proposals.ts [hunk 4]
```diff
@@ -0,0 +1,243 @@
...
+      option: MappedVotes[vote.vote],
+      votingPower,
+      tokenAmount,
+    };
+  });
+
+  // calculate participation summary
+  const formattedTotalWeight = abbreviateTokenAmount(
+    parseFloat(
+      Number(formatUnits(totalWeight, token.decimals)).toFixed(2)
```
Comment (WissenIstNacht) [inline packages/web-app/src/utils/proposals.ts:115]: Technically `abbreviateTokenAmount` also does rounding.

Either way, the number cast and then parse to string is not necessary, as `toFixed` already returns a string :)

// This happens in a couple of other places too, btw

File: packages/web-app/src/utils/proposals.ts [hunk 5]
```diff
@@ -0,0 +1,243 @@
...
+  );
+
+  const formattedParticipation = abbreviateTokenAmount(
+    parseFloat(
+      Number(formatUnits(weights, token.decimals)).toFixed(2)
+    ).toString()
+  );
+
+  const participationPercentage = parseFloat(
+    Big(Number(weights)).mul(100).div(Number(totalWeight)).toFixed(2)
```
Comment (WissenIstNacht) [inline packages/web-app/src/utils/proposals.ts:126]: I thought `div` would expect a `Big`? I.e., `Big(Number(totalWeight))`? (I might very well be wrong, though)

File: packages/web-app/src/containers/votingTerminal/index.tsx [hunk 6]
```diff
@@ -82,15 +82,15 @@ export const VotingTerminal: React.FC<VotingTerminalProps> = ({
...
-  const statusIcon = useMemo(
-    () =>
-      status === 'pending' || status === 'active' ? (
-        <IconClock className="text-info-500" />
-      ) : (
-        <IconInfo className="text-info-500" />
-      ),
-    [status]
-  );
+  const statusIcon = useMemo(() => {
```
Comment (WissenIstNacht) [inline packages/web-app/src/containers/votingTerminal/index.tsx]: Typically, functions that return components are themselves components :) And if you're really worried about performance, you can always wrap that component with memo.

**Review by WissenIstNacht — COMMENTED** [review]
Cheers man, lots of good things in here.

File: packages/web-app/src/pages/proposal.tsx (hunk 1 above)
Comment (Fabricevladimir) [reply packages/web-app/src/pages/proposal.tsx]: Fair point 👍🏾

File: packages/web-app/src/pages/proposal.tsx (hunk 2 above)
Comment (Fabricevladimir) [reply packages/web-app/src/pages/proposal.tsx]: Oof, residue from disabling subgraph check. Good catch

File: packages/web-app/src/utils/proposals.ts (hunk 3 above)
Comment (Fabricevladimir) [reply packages/web-app/src/utils/proposals.ts:87]: It is not. This is perhaps a naming thing. `totalWeight` is how many tokens total can vote based on when the proposal was created, whereas `weights` is the total of all tokens _actually voted_.

Also, it is being used for `formattedParticipation` and `participationPercentage` on lines 121 & 126 respectively

File: packages/web-app/src/utils/proposals.ts (hunk 4 above)
Comment (Fabricevladimir) [reply packages/web-app/src/utils/proposals.ts:115]: Ah, yes there is a reason for this. I might extract it into some sort of function so it's centralized. We want 2 digits after the decimal point, but only when there are significant digits and not zero ie (1.45 -> good, 1 -> good, 1.00 -> bad).

The workaround itself: 
1. `toFixed` returns a string with two digits after the decimal including the zeros ('5.00')
2. `parseFloat` drops the digits after the decimal if they are only zeros leaving us with 5

It's not an elegant solution honestly and perhaps can be improved on (even if only by extracting it into some sort of function), but for now it works hehehe

File: packages/web-app/src/utils/proposals.ts (hunk 5 above)
Comment (Fabricevladimir) [reply packages/web-app/src/utils/proposals.ts:126]: `div` actually expects a `BigSource` which itself is `number | string | Big`.

File: packages/web-app/src/utils/proposals.ts (hunk 4 above)
Comment (WissenIstNacht) [reply packages/web-app/src/utils/proposals.ts:115]: Oh, I see. Come to think of it, this could actually be added to the abbreviation function itself.

**Review by WissenIstNacht — APPROVED** [review]
👌🏼

**Pull request merged**